python/tskit/combinatorics.py

```diff
@@ -272,15 +270,55 @@ def split_polytomies(
         if tree.num_children(u) > 2:
             root = TreeNode.random_binary_tree(tree.children(u), rng)
             root.label = u
-            time = tree.time(u) - epsilon
-            stack = [(child, time) for child in root.children]
+            root_time = tree.time(u)
+            stack = [(child, root_time) for child in root.children]
             while len(stack) > 0:
-                node, time = stack.pop()
+                node, parent_time = stack.pop()
                 if node.label is None:
-                    node.label = tables.nodes.add_row(time=time)
+                    if epsilon is None:
+                        child_time = np.nextafter(parent_time, -np.inf)
+                    else:
+                        child_time = parent_time - epsilon
+                    node.label = tables.nodes.add_row(time=child_time)
+                else:
+                    assert len(node.children) == 0
+                    # This is a leaf node connecting back into the original tree
+                    child_time = tree.time(node.label)
+                if parent_time <= child_time:
+                    u = root.label
+                    min_child_time = min(tree.time(v) for v in tree.children(u))
+                    min_time = root_time - min_child_time
+                    message = (
+                        f"Cannot resolve the degree {tree.num_children(u)} "
+                        f"polytomy rooted at node {u} with minimum time difference "
+                        f"of {min_time} to the resolved leaves."
+                    )
+                    if epsilon is None:
+                        message += (
+                            " The time difference between nodes is so small that "
+                            "more nodes cannot be inserted between within the limits "
+                            "of floating point precision."
+                        )
+                    else:
+                        # We can also have parent_time == child_time if epsilon
```

Comment:
I'm not sure I understand this comment. If epsilon === branch_length and both are >0, then parent_time cannot be equal to child_time? Or to put it another way, if epsilon is too large, isn't parent_time > child_time guaranteed?

Reply:
The issue happens when we *exactly* use up all the available branch length in the original tree, and so there's nothing left for the last branch in the resolved bit. I've updated the comment, thanks.